src/background/services/deduplicator.ts

```diff
@@ -44,6 +49,10 @@ export class Deduplicator {
           return res;
         })
         .catch((err) => {
+          if (!cacheRejections) {
+            this.cache.delete(key);
+          }
+
```

Comment:
What do you think about this approach?

Note: `Promise.reject(err)` should replace `throw err`.
```suggestion
          if (cacheRejections) {
            this.cache.set(key, { promise: Promise.reject(err) });
          } else {
            this.cache.delete(key);
          }
          
          return Promise.reject(err);
```

CC @sidvishnoi @DarianM

Reply:
basically this would make `cacheRejections` false by default to all `dedupe` calls. i think it's right

Reply:
@DarianM Can we split changes to deduplicator to a separate PR and add unit tests to ensure it works as intended?

Reply:
@sidvishnoi sure. added #742 as depends on. let's do this first pronto